src/test/java/org/isf/medicalstockward/test/Tests.java

```diff
@@ -1342,6 +1345,29 @@ public void testMedicalWardHashCode() throws Exception {
 		assertThat(medicalWard.hashCode()).isEqualTo(hashCode);
 	}
 
+	@Test
+	public void testDeleteLastMovementWard() throws Exception {
+		int code = setupTestMovementWardWithMedicalWard(false);
+		Optional<MovementWard> lastMovementWard = movementWardIoOperationRepository.findById(code); 
+		assertThat(lastMovementWard).isPresent();
+		movWardBrowserManager.deleteLastMovementWard(lastMovementWard.get());
+		Optional<MovementWard> movement = movementWardIoOperationRepository.findById(code); 
+		assertThat(movement).isNotPresent();
+	}
+	
+	@Test
+	public void testDeleteLastMovementWardDenied() throws Exception {
+		int code = setupTestMovementWardWithMedicalWard(false);
+		MovementWard movementWard = movementWardIoOperationRepository.findById(code).orElse(null);
+		assertThat(movementWard).isNotNull();
+		Medical medical = movementWard.getMedical();
+		Ward ward = movementWard.getWard();
+		Lot lot = movementWard.getLot();
+		MovementWard lastMovementWard = new MovementWard(ward, lot, "newDescription", medical, 10.0, "newUnits");
+		movementWardIoOperationRepository.saveAndFlush(lastMovementWard);
+		assertThrows(OHServiceException.class, () -> movWardBrowserManager.deleteLastMovementWard(movementWard));
```

Comment:
I would continue the test by deleting the `lastMovementWard` (it should be ok) then adding a `newMovement` to another ward, then a `newMovement2` on that ward and then trying to delete the `newMovement`, wdyt?

Reply:
I think It should beOkay.

Reply:
ok, while you add this I'm going to test thoroughly